Src/VimCore/Interpreter_Interpreter.fs

```diff
@@ -807,6 +808,97 @@ type VimInterpreter
             if lineRange.Count > 1 then
                 _foldManager.CreateFold lineRange)
 
+    member x.RunNormal (lineRange: LineRangeSpecifier) input =
+        let transactionMap = System.Collections.Generic.Dictionary<ITextBuffer, ILinkedUndoTransaction>();
+        let modeSwitchMap = System.Collections.Generic.Dictionary<ITextBuffer, IVimBuffer>();

```

Comment:
I take back this bit of feedback.  I went back and looked at the RunMacro implementation and it reminded me why I built the maps this way.  My assertion about mapping between `ITextBuffer` and `IVimBuffer` is correct.  But in this case it's being tracked to `ILinkedUndoTransaction`.  That is definitely tied 1:1 with `ITextBuffer`, even if that value is shared with multiple `IVimBuffer`.  So `transactionMap` is correct here.  

I think `modeSwitchMap` could be just a `HashSet<IVimBuffer>` here.  But doesn't seem like a big deal.